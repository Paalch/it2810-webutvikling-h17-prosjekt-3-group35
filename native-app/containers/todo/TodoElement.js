import React from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import PropTypes from 'prop-types';

export default class TodoElement extends React.Component {
    render() {
        return (
            <View>
                <input
                    style={styles.todoElement}
                    type={'checkbox'}
                    onProgress={() => this.props.removeElement(this.props.text)}/>
                <label>{this.props.text}</label>
            </View>
        );
    }
}

TodoElement.propTypes = {
    text: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
    todoElement: {
        fontWeight: "normal",
    },
});

