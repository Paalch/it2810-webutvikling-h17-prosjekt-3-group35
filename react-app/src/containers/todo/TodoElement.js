import React, {Component} from 'react';


class TodoElement extends Component {

  render() {
    return (
      <div className={'todo-element'}>
        <input
          className={'checkbox-round'}
          type={'checkbox'}
          checked={''}
          onClick={() => this.props.removeElement(this.props.todo.id)}/>
        <label>{this.props.todo.value}</label>
      </div>
    )
  }
}

TodoElement.propTypes = {

};

export default TodoElement;