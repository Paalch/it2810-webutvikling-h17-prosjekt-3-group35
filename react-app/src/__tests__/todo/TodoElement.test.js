import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import TodoElement from '../../containers/todo/TodoElement'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
Enzyme.configure({ adapter: new Adapter() })

describe('Checks that TodoElement renders correctly', () => {

    //define todo
    const value = 2
    let testFunction = (test) => {return null}
    const todo = {value: value, removeElement: testFunction, id:0}

    it('renders correctly', () => {
        const rendered = renderer.create(
            <TodoElement todo={todo} />
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });

    it('Checks that the label gets the correct value' , () => {
        //Creates the wrapper, using shallow
        const wrapper = shallow(<TodoElement todo={todo}/>)
        expect(wrapper.find('label').text()).toBe(value.toString())

    })

    });