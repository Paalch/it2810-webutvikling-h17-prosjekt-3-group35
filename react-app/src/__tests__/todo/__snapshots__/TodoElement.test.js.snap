// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that TodoElement renders correctly renders correctly 1`] = `
<div
  className="todo-element"
>
  <input
    checked=""
    className="checkbox-round"
    onClick={[Function]}
    type="checkbox"
  />
  <label>
    2
  </label>
</div>
`;
